### PR TITLE
Rspec Testing Enumerables: Review

### PR DESCRIPTION
**Pull request by yigitm**
Add tests for all enumerable methods.

Please note that this is the second branch that does not show all commits since the previous branch deleted after re-naming this one.

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# rubocop:disable Style/SymbolProc
+
+require 'enumerables'
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:3]: Please, use `require_relative`

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      expect(my_array.my_each { |x| x }).to eq([1, 2, 3, 4, 5])
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'Prints each item index in array' do
+      expect(my_array.my_each_with_index { |x| x }).to eq([1, 2, 3, 4, 5])
+    end
+  end
+
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:19]: - Check if it returns an enumerator
- Check it when it received a bloc method
- I suggest you to **compare all your results with the original method**, in this case `each_with_index`

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  describe '#my_select' do
+    it 'Select items from array that covers given condition' do
+      expect(my_array.my_select { |x| x.even? }).to eq([2, 4])
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:24]: - Check if it returns an enumerator
- Check it when it received a bloc method
- I suggest you to compare all your results with the original method, in this case each_with_index

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  describe '#my_all?' do
+    it 'Return false if given condition is true for all elements' do
+      expect(my_array.my_all? { |x| x.odd? }).to eq(false)
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:30]: - Check it with block as parameters
- Check it with pattern, integer, String
-

File: spec/enumerables_spec.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    it 'Return true if any of the item cover given condition' do
+      expect(my_array.my_any? { |x| x.odd? }).to eq(true)
+    end
+  end
+
+  describe '#my_none?' do
+    it 'Return true if all items are not integer' do
+      expect(my_array.my_none? { |x| x.class != Integer }).to eq(true)
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:42]: - Check it with block as parameters
- Check it with pattern, integer, String
-

File: spec/enumerables_spec.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  describe '#my_count?' do
+    it 'Count the items of the given array' do
+      expect(my_array.my_count { |x| x }).to eq(5)
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:48]: - Count a value in an array
- Count the number the element of an array
- Count according to a block as parameter

File: spec/enumerables_spec.rb [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+  describe '#my_map' do
+    it 'Returns a new array with the result of the given condition' do
+      expect(my_array.my_map { |x| x * x }).to eq([1, 4, 9, 16, 25])
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:54]: - Take proc as argument
- Check if is an enumerator
-

File: spec/enumerables_spec.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+  describe '#my_inject' do
+    it 'Returns sum of all numbers and multiples by given value' do
+      expect(my_array.my_inject(3) { |x, y| x * y }).to eq(360)
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:60]: Add more tests

**Comment by visva-dev** [comment]
Implemented more test cases as asked by TSE.

**Pull request merged**